### DISCUSSION TITLE
compile_commands.json - how is this recognized

### DISCUSSION THREAD

**Discussion by fab6** [emacs-lsp/lsp-mode#3478] [Q&A] [answered]
Hi,

I created with `bear` a compilation database for a larger project 
[https://develop.openfoam.com/Development/openfoam](url)

This seemed to work fine.

The beginning of the files look like
![image](https://user-images.githubusercontent.com/3047695/164225332-503f56d6-dced-4990-bf0e-0194c10158b5.png)

In total I have 9735 lines; the file got initially created in the build directory; I copied it now to the project root directory.

Starting `lsp-mode` in one of the files of this project seem to work to some extend
![image](https://user-images.githubusercontent.com/3047695/164225724-5ab964be-1a86-486a-a659-6000c474c425.png)

Though it seems that the `compile_commands.json` file is not recognized, e.g. I am never able to get a definition, implementation or references

![image](https://user-images.githubusercontent.com/3047695/164226383-2dd615d2-41f5-41bf-a4ac-c203b556a470.png)

This is my main setup
``` lisp
(use-package lsp-mode
  :straight t
  :defer t
  :init
  (setq lsp-keymap-prefix "C-c l")
  (add-to-list 'exec-path fb/conda-env-home-directory)
  (setq lsp-prefer-flymake nil)
  :hook ((lsp-mode . lsp-enable-which-key-integration))
  :commands (lsp lsp-deferred)
  :config
  (add-hook 'c-mode-hook 'lsp)
  (add-hook 'c++-mode-hook 'lsp)
  (add-hook 'c-mode--hook #'lsp-clangd-c-enable)
  (add-hook 'c++-mode-hook #'lsp-clangd-c++-enable)
  (add-hook 'objc-mode-hook #'lsp-clangd-objc-enable)
  (setq lsp-clangd-binary-path fb/clangd-exe)
  (setq lsp-clients-clangd-executable fb/clangd-exe)
  (setq lsp-clients-fortls-executable fb/lsp-clients-fortls-executable)
  (setq lsp-clients-fortls-args fb/lsp-clients-fortls-args)
  (setq lsp-print-io t)
  (setq lsp-signature-auto-activate t)
  (setq lsp-signature-render-documentation t)
  (setq lsp-enable-symbol-highlighting t)
  (setq lsp-lens-enable t)
  (setq lsp-headerline-breadcrumb-enable t)

  (setq lsp-idle-delay 0.5
        lsp-enable-symbol-highlighting t
        lsp-enable-snippet nil  
        lsp-pyls-plugins-flake8-enabled t)

  (add-to-list 'lsp-language-id-configuration '(fortran-mode . "fortran"))
  (add-to-list 'lsp-language-id-configuration '(F90-mode . "fortran"))
  (lsp-register-custom-settings
   '(("pyls.plugins.pyls_mypy.enabled" t t)
     ("pyls.plugins.pyls_mypy.live_mode" nil t)
     ("pyls.plugins.pyls_black.enabled" t t)
     ("pyls.plugins.pyls_isort.enabled" t t)

     ;; Disable these as they're duplicated by flake8
     ("pyls.plugins.pycodestyle.enabled" nil t)
     ("pyls.plugins.mccabe.enabled" nil t)
     ("pyls.plugins.pyflakes.enabled" nil t)))
  )
```
I saw the discussion here and the guide to setup for c++, but it seems I am missing how to activate the recognition for the `compile_commands.json` file!?

[https://github.com/emacs-lsp/lsp-mode/discussions/2968](url)

Does anyone have a suggestion?
Thank you in advance!
Fab

**Comment by fab6** [comment]
Some further information using `lsp-doctor`
![image](https://user-images.githubusercontent.com/3047695/164284111-495dec5a-bed5-4d9a-8e18-a92fd6d8508d.png)
Not, sure, if the `error` is relevant for the `json` file handling...

**Comment by sebastiansturm** [comment]
do you get any output in the `*clangd*` or `*clangd:stderr*` buffers? If not, maybe add `--log=verbose` to `lsp-clients-clangd-args` and try again

**Reply by fab6** [reply]
Hi, thank you for the help! No, both are empty. I will check with your comments further and post any update.

**Comment by fab6** [comment, marked as answer]
It seem to work now... the compile_commands.json file was not complete. It has now roughly 150k lines. 
Thank you for your help!
